### PR TITLE
add frogger-game

### PR DESCRIPTION
**Pull request by usides**
### frogger-game

[Demo](https://usides.github.io/frontend-nanodegree-arcade-game/) |
[Code base](https://github.com/usides/frontend-nanodegree-arcade-game)

The code is submitted in a dedicated feature branch.
Please, review.

### CODE REVIEW

File: submissions/usides/frogger-game/app.js [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+      break;
+  }
+};
+
+Player.prototype.render = function () {
+  ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+};
+
+const player = new Player();
+const allEnemies = [new Enemy(1), new Enemy(2), new Enemy(3)];
```
Comment (OleksiyRudenko) [inline submissions/usides/frogger-game/app.js:117]: How about `const allEnemies = [1, 2, 3].map(row => new Enemy(row))`?
This way the code gets drier, it is clear what the argument passed over to enemy construction is about, and also allows to pass player instance in a single statement (if needed).

**Review by OleksiyRudenko — CHANGES_REQUESTED** [review]
Well done!
Quite a few improvements would be nioce to have. Please check the comments below

**Comment by OleksiyRudenko** [comment]
Not an essence of the task but it seems that the playfield is too big for a desktop/laptop FullHD screen and requires scrolling. Not a really nice UX.
Looks like canvas has upscaled for some reason. Please fix if you see the way to make it quick and also want to have a nice project in your portfolio.

**Comment by usides** [comment]
Honestly, I didn't understand what you mean in your previous comment. I didn't change CSS and HTML for this task. I don't have any scrolling bars and everything looks nice.

File: submissions/usides/frogger-game/app.js [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+Enemy.prototype.render = function () {
+  ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+};
+
+Enemy.prototype.checkCollision = function () {
+  if (
+    this.x >= player.x - conf.bugWidth &&
+    this.x <= player.x + conf.playerWidth &&
+    this.y == player.y
```
[the remark this answers is not in the document]
Comment (usides) [reply submissions/usides/frogger-game/app.js:43]: Done

File: submissions/usides/frogger-game/app.js (hunk 1 above)
Comment (usides) [reply submissions/usides/frogger-game/app.js]: Done

File: submissions/usides/frogger-game/app.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (OleksiyRudenko) [reply submissions/usides/frogger-game/app.js:43]: Still refers to global `player`. Simple test: change variable name in lines 116 & 118 and check if the app still works.

**Review by OleksiyRudenko — CHANGES_REQUESTED** [review]
One issue is still pending.

File: submissions/usides/frogger-game/app.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (usides) [reply submissions/usides/frogger-game/app.js]: Oh. I got it.

**Review by OleksiyRudenko — APPROVED** [review]
@usides well done!

**Pull request merged**